pkg/kubectl/polymorphichelpers/logsforobject.go

```diff
@@ -42,21 +42,94 @@ func logsForObject(restClientGetter genericclioptions.RESTClientGetter, object,
 	if err != nil {
 		return nil, err
 	}
-	return logsForObjectWithClient(clientset, object, options, timeout)
+	return logsForObjectWithClient(clientset, object, options, timeout, allContainers)
 }
 
 // this is split for easy test-ability
-func logsForObjectWithClient(clientset internalclientset.Interface, object, options runtime.Object, timeout time.Duration) (*rest.Request, error) {
+func logsForObjectWithClient(clientset internalclientset.Interface, object, options runtime.Object, timeout time.Duration, allContainers bool) ([]*rest.Request, error) {
 	opts, ok := options.(*coreinternal.PodLogOptions)
 	if !ok {
 		return nil, errors.New("provided options object is not a PodLogOptions")
 	}
 
 	switch t := object.(type) {
+	case *coreinternal.PodList:
```

Comment:
PodList is not supported originally.
Seems PodList does not have much relationship with allContainers.

Reply:
> PodList is not supported originally.
> Seems PodList does not have much relationship with allContainers.

It keeps the log gathering from being generic.  This fixes that and will force future updates to consider *all* resources, not just one type.